test/job_tests.cpp

```diff
@@ -33,50 +35,128 @@ BOOST_AUTO_TEST_CASE(get_job){
     free_client(client);
 }
 
+BOOST_AUTO_TEST_CASE(get_jobs){
+    ds3_client* client = get_client();
+    ds3_error* error = NULL;
+    ds3_request* request = NULL;
+
+    ds3_get_bucket_response* get_bucket_response = NULL;
+    ds3_bulk_response* bulk_response = NULL;
+    ds3_get_jobs_response* get_jobs_response = NULL;
 
+    const char* bucket1_name = "bucket_test_get_job1";
+    ds3_bulk_object_list* bucket1_object_list = NULL;
+    const char* bucket2_name = "bucket_test_get_job2";
+    ds3_bulk_object_list* bucket2_object_list = NULL;
+
+    printf("-----Testing Get Jobs-------\n");
+
+    /* create bucket1 with objects */
+    populate_with_objects(client, bucket1_name);
+    request = ds3_init_get_bucket(bucket1_name);
+    error = ds3_get_bucket(client, request, &get_bucket_response);
+    ds3_free_request(request);
+    handle_error(error);
+
+    // retain object_list for ds3_init_get_bulk
+    bucket1_object_list = ds3_convert_object_list(get_bucket_response->objects, get_bucket_response->num_objects);
+    ds3_free_bucket_response(get_bucket_response);
+
+    /* create bucket2 with objects */
+    populate_with_objects(client, bucket2_name);
+    request = ds3_init_get_bucket(bucket2_name);
+    error = ds3_get_bucket(client, request, &get_bucket_response);
+    ds3_free_request(request);
+    handle_error(error);
+
+    // retain object_list for ds3_init_get_bulk
+    bucket2_object_list = ds3_convert_object_list(get_bucket_response->objects, get_bucket_response->num_objects);
+    ds3_free_bucket_response(get_bucket_response);
+
+    /* init bulk get bucket1 */
+    request = ds3_init_get_bulk(bucket1_name, bucket1_object_list, NONE);
+    error = ds3_bulk(client, request, &bulk_response);
+    ds3_free_bulk_object_list(bucket1_object_list);
+    ds3_free_bulk_response(bulk_response);
+    ds3_free_request(request);
+    handle_error(error);
+
+    /* init bulk get bucket2 */
+    request = ds3_init_get_bulk(bucket2_name, bucket2_object_list, NONE);
+    error = ds3_bulk(client, request, &bulk_response);
+    ds3_free_bulk_object_list(bucket2_object_list);
+    ds3_free_bulk_response(bulk_response);
+    ds3_free_request(request);
+    handle_error(error);
+
+    /* GET jobs */
+    request = ds3_init_get_jobs();
+    error = ds3_get_jobs(client, request, &get_jobs_response);
+    ds3_free_request(request);
+    handle_error(error);
+
+    /* Assert valid response contents */
+    BOOST_CHECK(get_jobs_response != NULL);
+    uint8_t job_index;
+    for( job_index = 0; job_index < get_jobs_response->jobs_size; job_index++ ) {
+        bulk_response = get_jobs_response->jobs[job_index];
+        BOOST_CHECK( bulk_response->bucket_name->value != NULL);
+        BOOST_CHECK( bulk_response->job_id->value != NULL);
+        BOOST_CHECK( bulk_response->start_date->value != NULL);
+        BOOST_CHECK( bulk_response->user_id->value != NULL);
+        BOOST_CHECK( bulk_response->user_name->value != NULL);
+    }
```

Comment:
You should probably check to make sure that one of the jobs you started above is in the list using the job id.